cmd/crawler/main.go

```diff
@@ -2,10 +2,49 @@ package main
 
 import (
 	"fmt"
-
 	"github.com/wikidistance/wikidist/pkg/crawler"
 )
 
+type Result struct {
+	url   string
+	links []string
+}
+
 func main() {
-	fmt.Println(crawler.GetPageLinks("https://fr.wikipedia.org/wiki/Vilebrequin_(moteur)"))
+	queue := make(chan string, 10000)
+	defer close(queue)
+
+	results := make(chan Result, 3)
+
+	seen := make(map[string]bool)
+	graph := make(map[string][]string)
+
+	startUrl := "/wiki/Amauroclopius"
+	queue <- startUrl
+	seen[startUrl] = true
+
+	for i := 1; i <= 3; i++ {
+		go worker(queue, results)
+	}
+
+	for nCrawled := 0; nCrawled <= 100; nCrawled++ {
+		result := <-results
+		fmt.Println("got result", result.url, len(result.links))
+
+		graph[result.url] = result.links
+		for _, link := range result.links {
+			if _, already_seen := seen[link]; !already_seen {
```

Comment:
To check if something is in a map we usually do `if ok, _ := seen[link]; !ok {...}`. Moreover the standard way to create a set is to use a `map[string]struct{}` since the boolean is already encoded in the map itself and this is more efficient since a `struct{}{}` takes 0 byte in memory